command/agent.go

```diff
@@ -1047,3 +1050,19 @@ func (c *AgentCommand) handleMetrics() http.Handler {
 		}
 	})
 }
+
+func (c *AgentCommand) handleQuit() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Only handle POST/PUT requests
+		switch r.Method {
+		case http.MethodPost:
+		case http.MethodPut:
+		default:
+			w.WriteHeader(405)
+			return
+		}
+
+		c.logger.Debug("received quit request")
```

Comment:
Should we handle the case where `c.Logger` might be nil, assuming it is not required to be set somewhere else.

Reply:
I don't it's necessary, it's set early on in the Agent code path.